compiler/rustc_lint/src/unused.rs

```diff
@@ -157,6 +157,12 @@ impl<'tcx> LateLintPass<'tcx> for UnusedResults {
             },
             hir::ExprKind::AddrOf(..) => Some("borrow"),
             hir::ExprKind::Unary(..) => Some("unary operation"),
+            hir::ExprKind::MethodCall(ref method, ..) if fn_warned => match method.ident.name {
+                sym::add | sym::sub | sym::mul | sym::div | sym::rem | sym::neg => {
```

Comment:
This seems too broad to match by name alone. I think the right way to match names precisely is with a diagnostic item -- for example, see how [`is_dangerous_init`](https://github.com/rust-lang/rust/blob/001a77fac33f6560ff361ff38f661ff5f1c6bf85/compiler/rustc_lint/src/builtin.rs#L2418) does that.